resources/var/tmp/resources/init.groovy.d/scripts/init090NORMmavenautoinstall.groovy

```diff
@@ -46,32 +61,56 @@ def addMavenToInstallations(def installation) {
     Jenkins.instance.save()
 }
 
-def removeNonTargetM3Installations(def mavenName, def targetVersion) {
+// Checks if property has only one hudson.tasks.Maven$MavenInstaller installer and no other installers.
+// Other possible installers are hudson.tools.ZipExtractionInstaller,
+// hudson.tools.CommandInstaller or hudson.tools.BatchCommandInstaller.
+boolean propertyHasOneMavenInstallerOnly(def property) {
+    int numberOfMavenInstallers = 0
+    boolean noOtherInstallerDetected = true
+    property.installers.each { installer ->
```

Comment:
This for loop is unnecessary. Just check if the size is 1 and return false otherwise